packages/zora/src/Zora.ts

```diff
@@ -363,6 +348,66 @@ export const simulateMint = async (
   }
 }
 
+export const simulateV2Mint = async (
+  mint: MintIntentParams,
+  value: bigint,
+  client: PublicClient,
+  account?: Address,
+): Promise<SimulateContractReturnType> => {
+  const { contractAddress, tokenId, amount, recipient, referral } = mint
+
+  const from = account ?? DEFAULT_ACCOUNT
+
+  const _tokenId = await getNextTokenId(client, contractAddress, tokenId)
+
+  await checkBytecode(client, contractAddress)
+
+  const mintArgs = [
+    recipient ?? from,
+    amount,
+    contractAddress,
+    _tokenId,
+    referral ?? ZORA_DEPLOYER_ADDRESS,
+    '',
+  ]
+
+  const result = await client.simulateContract({
+    address: ZORA_TIMED_SALE_STRATEGY,
+    value,
+    abi: V2_MINT_ABI,
+    functionName: 'mint',
+    args: mintArgs,
+    account: from,
+    stateOverride: [
+      {
+        address: from,
+        balance: parseEther('100'),
+      },
+    ],
+  })
+  return result
+}
+
+export const simulateMint = async (
+  mint: MintIntentParams,
+  value: bigint,
+  account?: Address,
+  client?: PublicClient,
+): Promise<SimulateContractReturnType> => {
+  const _client =
+    client ??
+    (createPublicClient({
+      chain: chainIdToViemChain(mint.chainId),
+      transport: http(),
+    }) as PublicClient)
+
+  try {
+    return await simulateV2Mint(mint, value, _client, account)
+  } catch {
+    return await simulateV1Mint(mint, value, _client, account)
```

Comment:
this failover will obscure `simulateV2Mint` errors that aren't caused by the mint being v1 which could make debugging simulation errors more difficult. 

if there's a specific error we can check for to know if the mint is v1, that would be ideal. otherwise logging the error would give visibility into what's going on: 

```
  try {
    return await simulateV2Mint(mint, value, _client, account)
  } catch (error) {
    console.error("simulateV2Mint failed:", error)
    return await simulateV1Mint(mint, value, _client, account)
  }
  ```